lua/autorun/client/cl_devtools.lua

```diff
@@ -1,144 +1,171 @@
-﻿-- This file is subject to copyright - contact swampservers@gmail.com for more information.
+﻿--[[
+    Limitations:
+        - If a path has lua/STRUCTHERE eg. "lua/autorun/lua/weapons/test.lua" then
+            It'll pass the Struct into it, not sure if it's a bug or a feature.
+        - A file within a sub directory of a struct folder like: "lua/weapons/testEnt/mysubdir/lua"
+            Won't have the SWEP/struct table passed to it.
+
+]]--
+
+-- This file is subject to copyright - contact swampservers@gmail.com for more information.
 -- INSTALL: CINEMA
 SWAMP_DEV = SWAMP_DEV or {}
-SWAMP_DEV.isDev = SWAMP_DEV.isDev
+
+SWAMP_DEV.allowed = {
+    ["SteamID Here."] = true
+}
+
+-- Are we in development mode?
+SWAMP_DEV.enabled = false
+-- The delay on checking if we need to refresh files
 SWAMP_DEV.refreshDelay = 2
-SWAMP_DEV.itemStruct = SWAMP_DEV.itemStruct
+
+local currentItemStruct = nil
+
+--- This stores file times for certain root folder.
+--- FileNames[key] -> SaveTime[value]
 SWAMP_DEV.fileTimes = SWAMP_DEV.fileTimes or {}
+ 
+-- lua/STRUCT/*
 
-SWAMP_DEV.structTypes = {
+SWAMP_DEV.structEnvironments = {
     ["weapons"] = function(curClass)
-        SWEP = weapons.GetStored(curClass)
+        local found = weapons.GetStored(curClass)
+        return found and {SWEP=found} or nil
     end,
     ["entities"] = function(curClass)
-        ENT = scripted_ents.GetStored(curClass)
+        local found = scripted_ents.GetStored(curClass)
+        return found and {ENT=found} or nil
     end,
     ["effects"] = function(curClass)
-        local fxTab = effects.GetList()
-        local fx = "effects/" .. curClass
-
-        for _, v in ipairs(fxTab) do
-            if (v.Folder == fx) then
-                EFFECT = v
-
-                return
+        local allEffects = effects.GetList()
+        local targetEffect = "effects/" .. curClass
+
+        for _, effect in ipairs(allEffects) do
+            if effect.Folder == targetEffect then
+                return {
+                    EFFECT = effect
+                }
             end
         end
     end
 }
 
--- Returns a string to print
-local function refreshFile(path, filename)
-    local code = file.Read(path, "MOD")
-
-    -- current filename as class format
-    if SWAMP_DEV.itemStruct then
-        local parentFolder = string.gsub(string.sub(path, 1, -#filename - 2), ".*/", "")
 
-        -- if parent isn't a struct then use parent as class, else use filename
-        if (not SWAMP_DEV.structTypes[parentFolder]) then
-            SWAMP_DEV.structTypes[SWAMP_DEV.itemStruct](parentFolder)
-        else
-            SWAMP_DEV.structTypes[SWAMP_DEV.itemStruct](string.TrimRight(string.gsub(path, "^.*/", ""), ".lua"))
+local ROOT_DIRECTORY = "addons/contrib/"
+local refreshFolder
+local loadFile
```

Comment:
what's the point of putting these here and then making a `do ... end`, it's messy

Reply:
it minimises the scope that gets used and is an optimisation.